src/main/java/com/datatorrent/netlet/AbstractClient.java

```diff
@@ -49,7 +49,7 @@
 
   public boolean isConnected()
   {
-    return key.isValid() && ((SocketChannel)key.channel()).isConnected();
+    return key != null && key.isValid() && ((SocketChannel)key.channel()).isConnected();
```

Comment:
Yes, I am working on few test cases and will include the fix for the issue in the commit.